commit e40ca69691c1bb1f7c201c36b61e9cb1331e3d0f (HEAD -> master)
Author: Josh Wilsdon <jwilsdon@joyent.com>
Date:   2019-09-12T16:34:31-07:00 (4 weeks ago)
    
    TRITON-1895 prometheus mdata:execute should not go to maint when DNS is hooped
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/CHANGES.md b/CHANGES.md
index af668ee..7fa6141 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -1,5 +1,9 @@
 # triton-prometheus changes
 
+## 1.1.1
+
+TRITON-1895 Fix so mdata:execute doesn't go to maintenance when DNS is broken
+
 ## 1.1.0
 
 TRITON-1650 Add support for global zone discovery on Triton
diff --git a/boot/setup.sh b/boot/setup.sh
index 54c1979..664c9e1 100755
--- a/boot/setup.sh
+++ b/boot/setup.sh
@@ -231,9 +231,16 @@ else # "$FLAVOR" == "triton"
     sdc_log_rotation_add prometheus /var/svc/log/*prometheus*.log 1g
     sdc_log_rotation_setup_end
 
-    # Update the global_zones.json the first time
+    #
+    # Update the global_zones.json the first time.
+    #
+    # We disable errexit so that failure to update when external services
+    # (DNS) are broken does not abort the completion of setup.
+    #
+    set +o errexit
     ${ROOT_DIR}/bin/update_global_zones.sh \
         >>/var/log/update_global_zones.log 2>&1
+    set -o errexit
 
     sdc_setup_complete
 fi
diff --git a/package.json b/package.json
index 3bd4cea..84f6151 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
     "name": "triton-prometheus",
     "description": "Prometheus for TritonDC",
-    "version": "1.1.0",
+    "version": "1.1.1",
     "author": "Joyent (joyent.com)",
     "private": true,
     "repository": {
